tests/Feature/ApiRouteTest.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace Tests\Feature;
+
+use App\Models\User;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Http\UploadedFile;
+use Illuminate\Support\Facades\Storage;
+use Tests\TestCase;
+use Laravel\Sanctum\Sanctum;
+
+class ApiRouteTest extends TestCase
+{
+    // phpcs:disable PSR1.Methods.CamelCapsMethodName.NotCamelCaps
+
+    use RefreshDatabase;
+
+    /**
+     * Test non authenticated api/user route
+     *
+     *  @return void
+     */
+    public function test_nonAuthenticated_api_user_willRedirect()
+    {
+        $response = $this->get('/api/user');
+        $response->assertStatus(302);
+    }
+
+    /**
+     * Test the /api/upload route's get method
+     *
+     *  @return void
+     */
+    public function test_get_upload_wrongMethod()
+    {
+        $response = $this->get('/api/upload');
+        $response->assertStatus(405);
+    }
+
+    /**
+     * Test the /api/user route
+     *
+     *  @return void
+     */
+    // phpcs:ignore
+    public function test_user_returns_user_data()
+    {
+        $user = User::factory()->create();
+
+        Sanctum::actingAs($user);
+        $response = $this->get('/api/user');
+
+        $response->assertStatus(200)
+            ->assertJsonStructure([
+                'username',
+                'mw_userid',
+                'updated_at',
+                'created_at',
+                'id'
+            ]);
+    }
+
+    // TODO: re-work after merging of
+    // https://github.com/wmde/wikidata-mismatch-finder/pull/12
+
+    /**
+     * Test the /api/upload route
+     *
+     *  @return void
+     */
+    // phpcs:ignore
+    public function test_upload_file()
+    {
+
+        $user = User::factory()->create();
+
+        Storage::fake('mismatchFiles');
+        $file = UploadedFile::fake()->create('mismatchFile.csv');
+
+        Sanctum::actingAs($user);
+        
+        $response = $this->post('/api/upload', ['mismatchFile' => $file]);
+
```

Comment:
Looks great! We probably also want to check that the file was actually created in the fake storage after the upload, as part of this test. Take a look at the [laravel docs](https://laravel.com/docs/8.x/http-tests#testing-file-uploads) for a general example, and also at [this commit](https://github.com/wmde/wikidata-mismatch-finder/pull/12/commits/1bfdd36b50414cfabf71faf6a636d1b2796c5d2c) to see how to configure a storage "disk" you can later fake in the tests (inspired by a [stack overflow answer](https://stackoverflow.com/questions/54039755/adding-custom-path-for-storage-in-filesystems-php-laravel#54041057)).

Reply:
You are right, I thought at some point about adding it but forgot down the road.